File: backend/app/schemas/sessions.py
import uuid
from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel, Field

from app.config import settings
from app.services.scenario_service import TtsVoiceGender


class SessionStatus(StrEnum):
    ACTIVE = "active"
    COMPLETED = "completed"


class SessionCreate(BaseModel):
    setting: str = Field(min_length=1, max_length=settings.scenario_field_max_length)
    user_role: str = Field(min_length=1, max_length=settings.scenario_field_max_length)
    ai_role: str = Field(min_length=1, max_length=settings.scenario_field_max_length)
    goal: str = Field(min_length=1, max_length=settings.scenario_field_max_length)
    tts_voice: TtsVoiceGender = TtsVoiceGender.MALE


class MessageRead(BaseModel):
    id: uuid.UUID
    role: str
    content: str
    created_at: datetime


class SessionRead(BaseModel):
    id: uuid.UUID
    scenario_text: str
    setting: str | None = None
    user_role: str | None = None
    ai_role: str | None = None
    goal: str | None = None
    tts_voice: TtsVoiceGender = TtsVoiceGender.MALE
    status: SessionStatus
    created_at: datetime
    completed_at: datetime | None = None


class SessionDetailRead(SessionRead):
    messages: list[MessageRead] = Field(default_factory=list)

File: frontend/src/types/sessions.ts
export type TtsVoiceGender = 'male' | 'female'

export type SessionStatus = 'active' | 'completed'

export interface SessionRead {
  id: string
  scenario_text: string
  setting: string | null
  user_role: string | null
  ai_role: string | null
  goal: string | null
  tts_voice: TtsVoiceGender
  status: SessionStatus
  created_at: string
  completed_at: string | null
}

export interface MessageRead {
  id: string
  role: string
  content: string
  created_at: string
}

export interface SessionDetailRead extends SessionRead {
  messages: MessageRead[]
}

export interface SessionCreate {
  setting: string
  user_role: string
  ai_role: string
  goal: string
  tts_voice: TtsVoiceGender
}

export function sessionTitle(session: SessionRead): string {
  return session.setting ?? session.scenario_text
}

export function sessionSubtitle(session: SessionRead): string {
  if (session.setting && session.ai_role) {
    return `AIの役: ${session.ai_role}`
  }
  return session.scenario_text
}
